wbooks/src/redux/commentAction/reducers.js

```diff
@@ -0,0 +1,45 @@
+import { actionNames } from './actions';
+
+const initialState = {
+  commentList: [],
+  commentLoading: false
+};
+
+const date_compare = (date1, date2) => {
+  if (date1 > date2) return -1;
+  if (date1 < date2) return 1;
+  return 0;
+};
+
+const commentAction = (state = initialState, action) => {
+  switch (action.type) {
+    case actionNames.GET_COMMENTS_SUCCESFULL:
+      return {
+        ...state,
+        commentList: action.payload.comments.sort((comment, anotherComment) => date_compare(comment.created_at, anotherComment.created_at)).slice(0, 4)
+      };
+    case actionNames.GET_COMMENTS_FAILURE:
+      return {
+        state
+      };
```

Comment:
Be aware of that ! You're returning an object with a single field named `state` that contains our state. In other words, `{ state: state }` instead of `state`

Reply:
I do recommend to store the error. Otherwise this action shouldn't exist since it doesn't modify the state.